packages/@aws-cdk/aws-iam/test/test.legacy-resources.ts

```diff
@@ -0,0 +1,24 @@
+import cdk = require('@aws-cdk/cdk');
+import { Test } from 'nodeunit';
+import iam = require('../lib');
+
+export = {
+  'cloudformation XxxResource emits a warning'(test: Test) {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'test-stack');
+
+    new iam.cloudformation.UserResource(stack, 'LegacyResource', {
+      userName: 'MyUserName'
+    });
+
+    const out = app.synthesizeStack('test-stack');
+
+    const warnings = out.metadata['/test-stack/LegacyResource'].filter(md => md.type === 'aws:cdk:warning');
+    test.deepEqual(warnings.length, 1);
+    test.deepEqual(warnings[0].data,
+      'DEPRECATED: \"cloudformation.UserResource\" will be deprecated in the next release in ' +
+      'favor of \"CfnUser\" (see https://github.com/awslabs/aws-cdk/issues/878)');
```

Comment:
Is one release enough time for people to notice? I personally found myself upgrading every 5 or so versions because of the pace at which they happen.